### PR TITLE
feat: Improve node sdk with multiple features

### PR DESCRIPTION
**Pull request by Nsttt**
Closes PLA-23

### CODE REVIEW

File: biome.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
+  "organizeImports": {
+    "enabled": true
+  },
+  "formatter": {
+    "enabled": true,
+    "indentStyle": "tab",
```
Comment (AnderRV) [inline biome.json]: Any preference to use tabs over spaces (4) as we do in other repos?

File: src/index.ts [hunk 2]
```diff
@@ -1,106 +1,142 @@
...
-import * as fastq from 'fastq';
+import fastq from "fastq";
+import fetchRetry from "fetch-retry";
+import packageJson from "../package.json" assert { type: "json" };
 
-import { VERSION } from './version';
+const API_URL = "https://api.zenrows.com/v1/";
 
-const API_URL = 'https://api.zenrows.com/v1/';
+type HttpMethods = "GET" | "OPTIONS" | "POST" | "PUT" | "PATCH" | "DELETE";
```
Comment (AnderRV) [inline src/index.ts]: The API only supports `"GET" | "POST" | "PUT"`

File: biome.json (hunk 1 above)
Comment (Nsttt) [reply biome.json]: None. Got confused due to `.editorconfig`. Moved back to spaces !

File: src/index.ts (hunk 2 above)
Comment (AnderRV) [reply src/index.ts]: But we don't expose `put` method as we do with post. Plus we should test and check that the data is being sent properly.

Comment (Nsttt) [reply src/index.ts]: Done

Comment (Nsttt) [reply src/index.ts]: Are we interested in supporting it ? I can add it

Comment (AnderRV) [reply src/index.ts]: Less usage than POST or GET, so not a prio. You can leave as-is for now and create another Issue to add it.

**Review by AnderRV — CHANGES_REQUESTED** [review]
Please review the usage on both the readme and the examples directory. These are prepared to receive "axios like" responses and won't work as of now.
And the readme still mentions axios and AxiosPromise as the response.

Please also document briefly how to install, use and create versions when needed.

**Comment by AnderRV** [comment]
> Please review the usage on both the readme and the examples directory.

`/examples` directory should be updated with the new syntax too.

File: src/index.ts [hunk 3]
```diff
@@ -1,104 +1,155 @@
...
-	wait_for?: string;
-	wait?: number;
-	block_resources?: string;
-	window_width?: number;
-	window_height?: number;
-	device?: string;
-	original_status?: boolean;
-	custom_headers?: boolean;
-	[x: string]: unknown;
+export interface ZenrowsConfig {
```
Comment (AnderRV) [inline src/index.ts]: Typo? Or is there a reason to not use the proper name (`ZenRows`)?

File: src/index.ts [hunk 4]
```diff
@@ -1,104 +1,155 @@
...
+    { headers = {} }: { headers?: Headers } = {},
+  ): Promise<Response> {
+    return this.queue.push({ url, config, headers });
+  }
+
+  public post(
+    url: string,
+    config?: ZenrowsConfig,
+    { headers = {}, data = {} }: { headers?: Headers; data?: unknown } = {
+      headers: { "Content-Type": "application/x-www-form-urlencoded" },
```
Comment (AnderRV) [inline src/index.ts:85]: Is the default Content-Type needed now? Forced below as default in a different way.

File: src/index.ts (hunk 3 above)
Comment (Nsttt) [reply src/index.ts]: That's a 100% a Typo. Fixed!

File: src/index.ts (hunk 4 above)
Comment (Nsttt) [reply src/index.ts:85]: Yeah, can be removed as per my last change. Done as well.

**Review by AnderRV — APPROVED** [review]

**Comment by AnderRV** [comment]
@Nsttt can you please try to run the build script?

```
src/index.ts(84,5): error TS1016: A required parameter cannot follow an optional parameter.
```

image

**Pull request merged**